client/core/core.go

```diff
@@ -115,6 +121,25 @@ func (dc *dexConnection) markets() map[string]*Market {
 	return dc.marketMap
 }
 
+// getRegConfirms returns the number of confirmations received for the
+// dex registration or nil if the registration is completed
+func (dc *dexConnection) getRegConfirms() *uint32 {
+	dc.regConfMtx.RLock()
+	defer dc.regConfMtx.RUnlock()
+	if dc.regConfirms == regConfirmationsPaid {
+		return nil
+	}
+	return &dc.regConfirms
+}
```

Comment:
The `getRegConfirms` which returns an `uint32` pointer which can be used in the `core.Exchanges`. It will return the number of confirmations or `nil` if the registration has been completed. 
Notice that different from your suggestion I used a high number called `regConfirmationsPaid` to indicate that the confirmation is completed.
I could not use `0` because it just means that the registration is ongoing and no confirmation has been received so far. Using -1  requires converting the `dexConnection.regConfirms` to `int32` which hurts the semantic of the code and would also for me to convert the `Exchange` field to `int` at some point.